omnigibson/objects/object_base.py

```diff
@@ -82,8 +79,7 @@ def __init__(
         relative_prim_path = f"/{name}" if relative_prim_path is None else relative_prim_path
 
         # Store values
-        self.uuid = get_uuid(name) if uuid is None else uuid
-        assert len(str(self.uuid)) <= 8, f"UUID for this object must be at max 8-digits, got: {self.uuid}"
+        self._uuid = get_uuid(name, deterministic=True)
```

Comment:
What is the purpose of this UUID? It's annoying to me because these UUIDs are actually *not* unique in the sense that they are duplicated across scenes just like the object name - why do we need a separate identifier that serves exactly the same purpose as the name?

Reply:
it's for the sake of serializing the state, which, by construction, records the state as a flattened 1D numpy array. So we can't directly use strings.

Reply:
I'm opening a follow-up issue for renaming UUIDs into name hashes. I shouldnt hold up this PR on that. See #831.